### PR TITLE
#18946 Bring back base-type fields

### PR DESCRIPTION
**Pull request by dsilvam**
* Fields for each BaseType GraphQL Interface were brought back, but this time they are read from the ContentType inteface's `getRequiredField` method so the interface definitions can adapt to changes in them. 
* The map of Contentlet GraphQL Interface were moved to a seperate class to avoid circular dependencies between classes at initialization time. 
* Postman tests were included per each BaseType Interface. They check that the proper fields are returned.

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java [hunk 1]
```diff
@@ -110,32 +78,53 @@
...
         final Map<String, TypeFetcher> dotAssetFields = new HashMap<>(contentFields);
         interfaceTypes.put("DOTASSET", createInterfaceType(DOTASSET_INTERFACE_NAME, dotAssetFields, new ContentResolver()));
     }
 
-    public static Map<String, TypeFetcher> getContentFields() {
-        return contentFields;
+    private static void addBaseTypeFields(Map<String, TypeFetcher> baseTypeFields,
+            List<Field> requiredFormFields) {
+        for (final Field formField : requiredFormFields) {
+            if(!formField.fixed()) continue;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:123]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/ContentFields.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import com.dotcms.graphql.datafetcher.LanguageDataFetcher;
+import com.dotcms.graphql.datafetcher.SiteFieldDataFetcher;
+import com.dotcms.graphql.datafetcher.TitleImageFieldDataFetcher;
+import com.dotcms.graphql.datafetcher.UserDataFetcher;
+import com.dotcms.graphql.util.TypeUtil.TypeFetcher;
+import graphql.schema.GraphQLTypeReference;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ContentFields {
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/ContentFields.java:33]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java [hunk 3]
```diff
@@ -110,32 +78,53 @@
         interfaceTypes.put("CONTENT", createInterfaceType(CONTENT_INTERFACE_NAME, contentFields, new ContentResolver()));
 
         final Map<String, TypeFetcher> fileAssetFields = new HashMap<>(contentFields);
+        addBaseTypeFields(fileAssetFields, ImmutableFileAssetContentType.builder().name("dummy")
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:81]: Codacy found an issue: [The String literal "dummy" appears 7 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/ContentFields.java (hunk 2 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/ContentFields.java:33]: Codacy found an issue: [All methods are static.  Consider using a utility class instead. Alternatively, you could add a private constructor or make the class abstract to silence this warning.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java [hunk 4]
```diff
@@ -110,32 +79,55 @@
...
+                .build().requiredFields());
         interfaceTypes.put("DOTASSET", createInterfaceType(DOTASSET_INTERFACE_NAME, dotAssetFields, new ContentResolver()));
     }
 
-    public static Map<String, TypeFetcher> getContentFields() {
-        return contentFields;
+    private static void addBaseTypeFields(Map<String, TypeFetcher> baseTypeFields,
+            List<Field> requiredFormFields) {
+        for (final Field formField : requiredFormFields) {
+            if(!formField.fixed()) continue;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:126]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java [hunk 5]
```diff
@@ -110,32 +79,55 @@
         interfaceTypes.put("CONTENT", createInterfaceType(CONTENT_INTERFACE_NAME, contentFields, new ContentResolver()));
 
         final Map<String, TypeFetcher> fileAssetFields = new HashMap<>(contentFields);
+        addBaseTypeFields(fileAssetFields, ImmutableFileAssetContentType.builder().name("dummy")
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:82]: Codacy found an issue: [The String literal "dummy" appears 7 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:126]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:82]: Codacy found an issue: [The String literal "dummy" appears 7 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

**Review by freddyucv — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:126]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:82]: Codacy found an issue: [The String literal "dummy" appears 7 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/ContentFields.java [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        contentFields.put(MOD_DATE, new TypeFetcher(GraphQLString));
+        contentFields.put(TITLE, new TypeFetcher(GraphQLString));
+        contentFields.put(TITLE_IMAGE_KEY, new TypeFetcher(GraphQLTypeReference.typeRef("Binary"), new TitleImageFieldDataFetcher()));
+        contentFields.put(CONTENT_TYPE, new TypeFetcher(GraphQLString));
+        contentFields.put(BASE_TYPE, new TypeFetcher(GraphQLString));
+        contentFields.put(LIVE, new TypeFetcher(GraphQLBoolean));
+        contentFields.put(WORKING, new TypeFetcher(GraphQLBoolean));
+        contentFields.put(ARCHIVED_KEY, new TypeFetcher(GraphQLBoolean));
+        contentFields.put(LOCKED_KEY, new TypeFetcher(GraphQLBoolean));
+        contentFields.put("conLanguage", new TypeFetcher(GraphQLTypeReference.typeRef("Language"), new LanguageDataFetcher()));
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/ContentFields.java:51]: Nitpick:
No constant for this field.

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java [hunk 7]
```diff
@@ -62,15 +48,15 @@
...
+    final private Class<? extends ContentType> baseContentType;
 
     InterfaceType(final Class<? extends ContentType> baseContentType) {
         this.baseContentType = baseContentType;
     }
 
     public static Set<String> CONTENT_INTERFACE_FIELDS = new HashSet<>();
 
-    private static Map<String, GraphQLInterfaceType> interfaceTypes = new HashMap<>();
+    final private static Map<String, GraphQLInterfaceType> interfaceTypes = new HashMap<>();
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:59]: Nitpick: to match every other field, can we declare it in the same order: `private static final...`

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java]: Done

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:126]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:82]: Codacy found an issue: [The String literal "dummy" appears 7 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:126]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:82]: Codacy found an issue: [The String literal "dummy" appears 7 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:126]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:82]: Codacy found an issue: [The String literal "dummy" appears 7 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 4 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:126]: Codacy found an issue: [This statement should have braces](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

File: dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java (hunk 5 above)
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/graphql/InterfaceType.java:82]: Codacy found an issue: [The String literal "dummy" appears 7 times in this file; the first occurrence is on line 81](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5939440)

**Pull request merged**